Deprecation errors compiling process on MacOS

```
$ autoreconf -i
$ cabal build all
Build profile: -w ghc-8.10.7 -O1
In order, the following will be built (use -v for more details):
 - process-1.6.15.0 (lib:process) (first run)
Configuring process-1.6.15.0...
configure: WARNING: unrecognized options: --with-compiler
checking for gcc... /usr/bin/gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether /usr/bin/gcc accepts -g... yes
checking for /usr/bin/gcc option to enable C11 features... none needed
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for vfork.h... no
checking for pid_t... yes
checking for fork... yes
checking for vfork... yes
checking for working fork... yes
checking for working vfork... (cached) yes
checking for signal.h... yes
checking for sys/wait.h... yes
checking for fcntl.h... yes
checking for setitimer... yes
checking for sysconf... yes
checking for execvpe... no
checking for spawn.h... yes
checking for posix_spawnp... yes
checking for posix_spawn_file_actions_addchdir... no
checking for /usr/bin/gcc options needed to detect all undeclared functions... none needed
checking whether POSIX_SPAWN_SETSID is declared... yes
checking whether POSIX_SPAWN_SETSID_NP is declared... no
checking whether POSIX_SPAWN_SETPGROUP is declared... yes
checking whether posix_spawn reports errors sensibly... yes
checking value of SIG_DFL... 0
checking value of SIG_IGN... 1
configure: creating ./config.status
config.status: creating include/HsProcessConfig.h
configure: WARNING: unrecognized options: --with-compiler
Preprocessing library for process-1.6.15.0..
Building library for process-1.6.15.0..
[1 of 5] Compiling System.Process.Common ( System/Process/Common.hs, /Users/jky/wrk/iohk/process/dist-newstyle/build/aarch64-osx/ghc-8.10.7/process-1.6.15.0/build/System/Process/Common.o, /Users/jky/wrk/iohk/process/dist-newstyle/build/aarch64-osx/ghc-8.10.7/process-1.6.15.0/build/System/Process/Common.dyn_o )
[2 of 5] Compiling System.Process.Posix ( System/Process/Posix.hs, /Users/jky/wrk/iohk/process/dist-newstyle/build/aarch64-osx/ghc-8.10.7/process-1.6.15.0/build/System/Process/Posix.o, /Users/jky/wrk/iohk/process/dist-newstyle/build/aarch64-osx/ghc-8.10.7/process-1.6.15.0/build/System/Process/Posix.dyn_o )
[3 of 5] Compiling System.Process.Internals ( System/Process/Internals.hs, /Users/jky/wrk/iohk/process/dist-newstyle/build/aarch64-osx/ghc-8.10.7/process-1.6.15.0/build/System/Process/Internals.o, /Users/jky/wrk/iohk/process/dist-newstyle/build/aarch64-osx/ghc-8.10.7/process-1.6.15.0/build/System/Process/Internals.dyn_o )
[4 of 5] Compiling System.Process   ( System/Process.hs, /Users/jky/wrk/iohk/process/dist-newstyle/build/aarch64-osx/ghc-8.10.7/process-1.6.15.0/build/System/Process.o, /Users/jky/wrk/iohk/process/dist-newstyle/build/aarch64-osx/ghc-8.10.7/process-1.6.15.0/build/System/Process.dyn_o )
[5 of 5] Compiling System.Cmd       ( System/Cmd.hs, /Users/jky/wrk/iohk/process/dist-newstyle/build/aarch64-osx/ghc-8.10.7/process-1.6.15.0/build/System/Cmd.o, /Users/jky/wrk/iohk/process/dist-newstyle/build/aarch64-osx/ghc-8.10.7/process-1.6.15.0/build/System/Cmd.dyn_o )

cbits/posix/fork_exec.c:148:15: error:
     warning: 'vfork' is deprecated: Use posix_spawn or fork [-Wdeprecated-declarations]
        int pid = myfork();
                  ^
    |
148 |     int pid = myfork();
    |               ^

cbits/posix/fork_exec.c:32:16: error:
     note: expanded from macro 'myfork'
   |
32 | #define myfork vfork
   |                ^
#define myfork vfork
               ^

/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/unistd.h:604:1: error:
     note: 'vfork' has been explicitly marked deprecated here
    |
604 | __deprecated_msg("Use posix_spawn or fork")
    | ^
__deprecated_msg("Use posix_spawn or fork")
^

/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:208:48: error:
     note: expanded from macro '__deprecated_msg'
            #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                          ^
    |
208 |         #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
    |                                                ^
1 warning generated.

cbits/posix/fork_exec.c:148:15: error:
     warning: 'vfork' is deprecated: Use posix_spawn or fork [-Wdeprecated-declarations]
        int pid = myfork();
                  ^
    |
148 |     int pid = myfork();
    |               ^

cbits/posix/fork_exec.c:32:16: error:
     note: expanded from macro 'myfork'
   |
32 | #define myfork vfork
   |                ^
#define myfork vfork
               ^

/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/unistd.h:604:1: error:
     note: 'vfork' has been explicitly marked deprecated here
    |
604 | __deprecated_msg("Use posix_spawn or fork")
    | ^
__deprecated_msg("Use posix_spawn or fork")
^

/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:208:48: error:
     note: expanded from macro '__deprecated_msg'
            #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                          ^
    |
208 |         #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
    |                                                ^
1 warning generated.
```

--- a/cbits/posix/fork_exec.c
+++ b/cbits/posix/fork_exec.c
@@ -29,15 +29,9 @@
 #include <signal.h>
 #endif
 
-#if defined(HAVE_VFORK_H)
-#include <vfork.h>
-#endif
-
 #include <Rts.h>
 
-#if defined(HAVE_WORKING_VFORK)
-#define myfork vfork
-#elif defined(HAVE_WORKING_FORK)
+#if defined(HAVE_WORKING_FORK)
 #define myfork fork
 // We don't need a fork command on Windows
 #else
@@ -152,7 +146,7 @@ do_spawn_fork (char *const args[],
     }
 #endif
 
-    int pid = myfork();
+    int pid = fork();
     switch(pid)
     {
     case -1:
@@ -164,10 +158,6 @@ do_spawn_fork (char *const args[],
         return -1;
 
     case 0:
-        // WARNING! We may now be in the child of vfork(), and any
-        // memory we modify below may also be seen in the parent
-        // process.
-
         close(forkCommunicationFds[0]);
         fcntl(forkCommunicationFds[1], F_SETFD, FD_CLOEXEC);
 

--- a/changelog.md
+++ b/changelog.md
@@ -3,7 +3,8 @@
 ## unreleased
 
 * Fix deadlock when waiting for process completion and process jobs [#273](https://github.com/haskell/process/issues/273)
-* Support delegate_ctlc on Windows. [#278](https://github.com/haskell/process/pull/278)
+* Support `delegate_ctlc` on Windows. [#278](https://github.com/haskell/process/pull/278)
+* Drop support for `vfork` [#261](https://github.com/haskell/process/pull/261)
 
 ## 1.6.17.0 *February 2023*
 

--- a/configure.ac
+++ b/configure.ac
@@ -7,7 +7,7 @@ AC_CONFIG_HEADERS([include/HsProcessConfig.h])
 
 AC_PROG_CC
 
-dnl ** Working vfork?
+dnl ** Working fork?
 AC_FUNC_FORK
 
 # check for specific header (.h) files that we are interested in

